src/tools/run-make-support/src/lib.rs

```diff
@@ -1,16 +1,28 @@
 use std::env;
 use std::path::{Path, PathBuf};
 use std::process::{Command, Output};
-
+mod rustdoc;
+pub use rustdoc::rustdoc;
 pub use wasmparser;
 
+pub fn add_host_rpath_env(cmd: &mut Command) {
+    let ld_lib_path_envvar = env::var("LD_LIB_PATH_ENVVAR").unwrap();
+    let ld_lib_path_value = env::var(&ld_lib_path_envvar).unwrap();
+
+    let temp = env::var("TMPDIR").unwrap();
+    let host_rpath_dir = env::var("HOST_RPATH_DIR").unwrap();
+
+    cmd.env(ld_lib_path_envvar, format!("{temp}:{host_rpath_dir}:{ld_lib_path_value}"));
```

Comment:
AFAIK, including publicly-writable directories (like /tmp) into PATH-like environment variables can be dangerous. I think for test purposes it's probably not too bad, but do we actually need to do that? Can we avoid putting things into /tmp?

(Perhaps TMPDIR here is set by bootstrap in a way that isn't /tmp?)

Reply:
`$TMPDIR` isn't set by run-make (it's set in bootstrap, I need to go back to check where it's being set and what it's being set to), so this seems to just preserve whatever tools.mk was doing.

Reply:
Yeah, `$TMPDIR` here isn't `/tmp`, it's

https://github.com/rust-lang/rust/blob/9023f908cfbe7a475f369717a61cb8eb865cfd25/src/tools/compiletest/src/runtest.rs#L3730-L3734